### PR TITLE
JBPM-7786 - Provide more detailed messages and UI about exceptions

### PR DESCRIPTION
**Pull request by cristianonicolai**
- Example missing Kie Server role
missing_kie_server_role

- Example missing Process capability
missing_capability

- Example when no server is connected
no_server

### CODE REVIEW

**Comment by jhrcek** [comment]
@cristianonicolai will check this right after I'm done with EAP 7.2 testing.
Jenkins execute full downstream build

**Comment by jhrcek** [comment]
Jenkins execute full downstream build

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/test/java/org/jbpm/workbench/ht/client/editors/taskslist/AbstractTaskListPresenterTest.java [hunk 1]
```diff
@@ -48,6 +50,9 @@
 import org.jbpm.workbench.ht.service.TaskService;
 import org.junit.Before;
 import org.junit.Test;
+import org.kie.server.controller.api.model.spec.Capability;
+import org.kie.server.controller.api.model.spec.ServerTemplate;
+import org.kie.workbench.common.workbench.client.error.DefaultWorkbenchErrorCallback;
```
Comment (nmirasch) [inline jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/test/java/org/jbpm/workbench/ht/client/editors/taskslist/AbstractTaskListPresenterTest.java]: DefaultWorkbenchErrorCallback and ClientRuntimeError imports are not used, maybe they can be removed

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/test/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListPresenterTest.java [hunk 2]
```diff
@@ -739,19 +749,6 @@ public void testOpenTaskView() {
                      captor.getAllValues().get(1).getIdentifier());
     }
 
-    @Test
-    public void testCreateDataSetProcessInstanceCallbackOnError() {
-        final ProcessInstanceListPresenter spy = spy(presenter);
-        final ClientRuntimeError error = new ClientRuntimeError("");
```
Comment (nmirasch) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/test/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListPresenterTest.java]: Regarding ClientRuntimeError class is not used, maybe the import could be removed

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/test/java/org/jbpm/workbench/pr/client/editors/instance/log/ProcessInstanceLogPresenterTest.java [hunk 3]
```diff
@@ -37,22 +40,19 @@
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.kie.workbench.common.workbench.client.error.DefaultWorkbenchErrorCallback;
 import org.mockito.ArgumentCaptor;
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
+import org.mockito.Spy;
 import org.mockito.invocation.InvocationOnMock;
 import org.mockito.stubbing.Answer;
```
Comment (nmirasch) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/test/java/org/jbpm/workbench/pr/client/editors/instance/log/ProcessInstanceLogPresenterTest.java]: Answer and ErrorPopupPresenter imports are not used, maybe they could be removed

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/test/java/org/jbpm/workbench/pr/client/editors/instance/log/ProcessInstanceLogPresenterTest.java [hunk 4]
```diff
@@ -326,11 +333,11 @@ public void processInstaceSelectionDefaultFiltersAdditionTest() {
         verify(view,
                times(2)).addActiveFilter(captor.capture());
 
-        final ActiveFilterItem nodeTypefilterItem = captor.getAllValues().get(0);
-        assertNotNull(nodeTypefilterItem);
+        final ActiveFilterItem nodeTypeFilterItem = captor.getAllValues().get(0);
```
Comment (nmirasch) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/test/java/org/jbpm/workbench/pr/client/editors/instance/log/ProcessInstanceLogPresenterTest.java]: good catch! thanks!

File: jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/test/java/org/jbpm/workbench/es/client/editors/requestlist/RequestListPresenterTest.java [hunk 5]
```diff
@@ -58,6 +62,7 @@
 import org.uberfire.mvp.Commands;
 import org.uberfire.mvp.PlaceRequest;
 
+import static java.util.Collections.*;
 import static org.jbpm.workbench.es.client.editors.util.JobUtils.createRequestSummary;
 import static org.jbpm.workbench.es.model.ExecutionErrorDataSetConstants.EXECUTION_ERROR_LIST_DATASET;
```
Comment (nmirasch) [inline jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/test/java/org/jbpm/workbench/es/client/editors/requestlist/RequestListPresenterTest.java]: Not related with the PR changes but this EXECUTION_ERROR_LIST_DATASET could be removed :)

Comment (cristianonicolai) [reply jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/test/java/org/jbpm/workbench/es/client/editors/requestlist/RequestListPresenterTest.java]: +1 somehow my IntelliJ is not picking up all of these unused imports :/

File: jbpm-wb-common/jbpm-wb-common-api/src/main/java/org/jbpm/workbench/common/events/ServerTemplateSelected.java [hunk 6]
```diff
@@ -52,21 +53,21 @@ public boolean equals(Object o) {
 
         ServerTemplateSelected that = (ServerTemplateSelected) o;
 
-        return serverTemplateId.equals(that.serverTemplateId);
+        return serverTemplate.getId().equals(that.serverTemplate.getId());
```
[the remark this answers is not in the document]
Comment (cristianonicolai) [reply jbpm-wb-common/jbpm-wb-common-api/src/main/java/org/jbpm/workbench/common/events/ServerTemplateSelected.java]: Thanks, I improved the null handling here.

**Pull request merged**